daml-lf/archive/BUILD.bazel

```diff
@@ -39,21 +39,22 @@ load("@rules_pkg//:pkg.bzl", "pkg_tar")
             name = "daml_lf_%s_archive_proto" % version,
             srcs = [":daml_lf_%s_archive_proto_srcs" % version],
             javadoc_root_packages = ["com.daml.daml_lf_%s" % mangled_version],
-            maven_artifact_prefix = "daml-lf-%s-archive" % version,
+            maven_artifact_prefix = "daml-lf-%s-archive" % short_version,
             maven_group = "com.daml",
             strip_import_prefix = "src/%s/protobuf" % directory,
             visibility = ["//visibility:public"],
         ),
     ]
     for version in PROTO_LF_VERSIONS
-    for mangled_version in [mangle_for_java(version)]
-    for directory in ["main" if version == "dev" else "stable"]
+    for short_version in ["dev" if version == "1.dev" else version]
+    for mangled_version in [mangle_for_java(short_version)]
+    for directory in ["main" if version == "1.dev" else "stable"]
 ]
 
 # FIXME(JM): Clean this up
 genrule(
     name = "daml_lf_dev_archive_haskell_proto_src",
```

Comment:
Haskell does not like digit so we cannot rename it to 
`daml_lf_1.dev_archive_haskell_proto_src` nor `daml_lf_1_dev_archive_haskell_proto_src`.

In this case, we could alternatively drop the `dev` completely.